pkg/chunked/internal/compression.go

```diff
@@ -167,13 +170,41 @@ func WriteZstdChunkedManifest(dest io.Writer, outMetadata map[string]string, off
 		return err
 	}
 
+	var compressedTarSplitBuffer bytes.Buffer
+	compressTarSplitWriter, err := ZstdWriterWithLevel(&compressedTarSplitBuffer, level)
+	if err != nil {
+		return err
+	}
+	if _, err := compressTarSplitWriter.Write(tarSplitData); err != nil {
+		compressTarSplitWriter.Close()
+		return err
+	}
+	if err := compressTarSplitWriter.Close(); err != nil {
+		return err
+	}
+	compressedTarSplit := compressedTarSplitBuffer.Bytes()
+
+	tarSplitDigester := digest.Canonical.Digester()
+	tarSplitChecksum := tarSplitDigester.Hash()
+	if _, err := tarSplitChecksum.Write(compressedTarSplit); err != nil {
+		return err
+	}
+	outMetadata[TarSplitChecksumKey] = tarSplitDigester.Digest().String()
+	tarSplitOffset := manifestOffset + uint64(len(compressedManifest)) + 8
+	outMetadata[TarSplitInfoKey] = fmt.Sprintf("%d:%d:%d", tarSplitOffset, len(compressedTarSplit), len(tarSplitData))
+	if err := appendZstdSkippableFrame(dest, compressedTarSplit); err != nil {
+		return err
+	}
+
 	// Store the offset to the manifest and its size in LE order
 	manifestDataLE := make([]byte, FooterSizeSupported)
 	binary.LittleEndian.PutUint64(manifestDataLE, manifestOffset)
-	binary.LittleEndian.PutUint64(manifestDataLE[8:], uint64(len(compressedManifest)))
-	binary.LittleEndian.PutUint64(manifestDataLE[16:], uint64(len(manifest)))
-	binary.LittleEndian.PutUint64(manifestDataLE[24:], uint64(ManifestTypeCRFS))
-	copy(manifestDataLE[32:], ZstdChunkedFrameMagic)
+	binary.LittleEndian.PutUint64(manifestDataLE[8*1:], uint64(len(compressedManifest)))
+	binary.LittleEndian.PutUint64(manifestDataLE[8*2:], uint64(len(manifest)))
+	binary.LittleEndian.PutUint64(manifestDataLE[8*3:], uint64(ManifestTypeCRFS))
+	binary.LittleEndian.PutUint64(manifestDataLE[8*4:], uint64(tarSplitOffset))
+	binary.LittleEndian.PutUint64(manifestDataLE[8*5:], uint64(len(compressedTarSplit)))
```

Comment:
The reader seems to consume only the `TarSplitInfoKey` etc. annotation. Why does this need to be written into the footer as well?